examples/async_trace.cpp

```diff
@@ -74,26 +80,28 @@ int main() {
 
   auto start = steady_clock::now();
 
-  sync_wait(transform(
-      when_all(
+  sync_wait(
+    transform(
+      defer([&]{
```

Comment:
Why was defer() required here?
There shouldn't be a need for repeatability of the operation if just passing to sync_wait() without a repeat/retry of some sort.

Reply:
I narrowed down the scope at which to put this `defer()` to get rid of a compile error.

Reply:
I think I found the cause of this issue.

The transform() sender's connect() methods were not SFINAE-friendly and so when trying to call connect() on an rvalue it was still trying to instantiate the const-lvalue-ref connect() overload. Once transform() and when_all()'s connect() methods were modified to SFINAE out if input senders were not connectable then this defer() is no longer required.

See #106